src/MySqlConnector/Protocol/Payloads/HandshakeResponse41Payload.cs

```diff
@@ -5,11 +5,8 @@ namespace MySqlConnector.Protocol.Payloads;
 
 internal static class HandshakeResponse41Payload
 {
-	private static ByteBufferWriter CreateCapabilitiesPayload(ProtocolCapabilities serverCapabilities, ConnectionSettings cs, bool useCompression, CharacterSet characterSet, ProtocolCapabilities additionalCapabilities = 0)
-	{
-		var writer = new ByteBufferWriter();
-
-		var clientCapabilities = (ProtocolCapabilities.Protocol41 |
+	public static ProtocolCapabilities CreateClientCapabilities(ProtocolCapabilities serverCapabilities, ConnectionSettings cs, bool useCompression, ProtocolCapabilities additionalCapabilities = 0) =>
+		(ProtocolCapabilities.Protocol41 |
```

Comment:
I can't tell if this was a required change, or just a refactoring done along the way?

Either way, some change here seems to be causing a "Bad handshake" error.